notebooks/C_CNN_Training_and_Validation.ipynb

```diff
@@ -1,602 +1,602 @@
+    "\n",
+    "**Prior to calling the function to create the TFRecods files:**\n",
+    "\n",
+    "You need to manually drag the annotation_XXX.zip files into the newly created folders. If you are working in the Google Colab environment, click on the folder icon at the left-side dashboard, which should now contain the 'logs', 'train' and 'test' directories. They should be empty until you drag your annotation files into them.\n",
+    "\n",
+    "Once the files have been imported, run the following cell:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "Gjsrmn4Gm1qG"
+   },
+   "outputs": [],
+   "source": [
+    "LABELS = create_tf_record(TRAIN_PATH, TRAIN_FILE, use_flagged=not BOUNDARY_AUGMENTATION)\n",
+    "_ = create_tf_record(TEST_PATH, TEST_FILE, use_flagged=not BOUNDARY_AUGMENTATION)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "NrLhVmunm1qG"
+   },
+   "source": [
+    "### Create a simple CNN for classification:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "KGgzMUfmm1qG"
+   },
+   "outputs": [],
+   "source": [
+    "img = K.layers.Input(shape=(64, 64, 1))\n",
+    "x = Encoder2D(layers=[8, 16, 32, 64, 128])(img)\n",
+    "x = K.layers.Flatten()(x)\n",
+    "x = K.layers.Dense(256, activation=\"relu\")(x)\n",
+    "x = K.layers.Dropout(0.2)(x)\n",
+    "logits = K.layers.Dense(len(LABELS), activation=\"linear\")(x)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "ECXUoYTMm1qG"
+   },
+   "outputs": [],
+   "source": [
+    "model = K.Model(inputs=img, outputs=logits)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "nO-zelEnm1qH"
+   },
+   "outputs": [],
+   "source": [
+    "model.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "zqYuOJzmm1qH"
+   },
+   "source": [
+    "### Set-up some augmentations to be used while training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "RolEOzdkm1qH"
+   },
+   "outputs": [],
+   "source": [
+    "@augmentation_label_handler\n",
+    "def normalize(img):\n",
+    "    img = tf.image.per_image_standardization(img)\n",
+    "    # clip to 4 standard deviations\n",
+    "    img = tf.clip_by_value(img, -4., 4.)\n",
+    "    tf.debugging.check_numerics(img, \"Image contains NaN\")\n",
+    "    return img"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "OaAOKscam1qH"
+   },
+   "outputs": [],
+   "source": [
+    "@augmentation_label_handler\n",
+    "def augment(img):\n",
+    "    if BOUNDARY_AUGMENTATION:\n",
+    "        # this will randomly simulate the cropping that occurs at the edge of\n",
+    "        # an image volume\n",
+    "\n",
+    "        vignette = np.ones((model.layers[0].input_shape[0][1], \n",
```

Comment:
then finally, we can use:
`vignette = np.ones(INPUT_SHAPE, dtype=np.float32)`

probably also changing the `30` in the random width to `INPUT_SHAPE // 2` while we're about it